### PR TITLE
DCS-1685 add record a body scan page, controller and validation

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: server/middleware/validation/bodyScanValidation.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    return [{ text: 'Year must include 4 numbers', href: '#another-date' }]
+
+  if (!isValidDate(d, m, y)) return [{ text: 'Date must be a real date', href: '#another-date' }]
+
+  if (date === 'another-date' && isFutureDate(d, m, y))
+    return [{ text: 'Enter a date in the past', href: '#another-date' }]
+
+  return !reason ? [{ text: 'Select a reason for the body scan', href: '#reason' }] : []
+}
+export default BodyScanValidator
```
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/middleware/validation/bodyScanValidation.ts:36]: Ah ok

[the remark this answers is not in the document]
Comment (andrewrlee) [reply server/middleware/validation/bodyScanValidation.ts:36]: might be clearer to split off methods out a separate method for the another date validation ?

File: server/middleware/validation/bodyScanValidation.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  if (date === 'another-date' && missingFieldNames.length)
+    return [{ text: `Date must include a ${message}`, href: `#another-date-${missingFieldNames[0]}` }]
+
+  const YEAR_REGEX = /^\d{4}$/
+
+  if (date === 'another-date' && !YEAR_REGEX.test(y))
+    return [{ text: 'Year must include 4 numbers', href: '#another-date' }]
+
+  if (!isValidDate(d, m, y)) return [{ text: 'Date must be a real date', href: '#another-date' }]
```
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/middleware/validation/bodyScanValidation.ts:29]: Sorted in next commit

File: server/views/pages/recentArrivals/bodyScan.njk [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                                name: "year",
+                                value: data.year,
+                                attributes: {'data-qa': 'another-date-year'}
+                                }
+                            ]
+                        }) }}
+                    {% endset -%}
+
+                {{ govukRadios({
+                    name: "date",
```
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/views/pages/recentArrivals/bodyScan.njk:60]: Will be sorted in next commit

File: server/middleware/validation/bodyScanValidation.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    return [{ text: 'Select a date for the body scan', href: '#date' }]
+  }
+
+  const missingFieldNames = zip(fields, [day, month, year])
+    .map(([field, value]) => !value && field)
+    .filter(Boolean)
+
+  const message = missingFieldNames.join(' and ')
+
+  if (date === 'another-date' && missingFieldNames.length)
```
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/middleware/validation/bodyScanValidation.ts:21]: Will be sorted in next commit

File: server/middleware/validation/bodyScanValidation.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (louisenorris) [reply server/middleware/validation/bodyScanValidation.ts:36]: Will be sorted in the next commit

**Review by davidthomas22 — APPROVED** [review]
LGTM

**Pull request merged**